miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/workflow/DefaultWorkflowManager.java

```diff
@@ -80,7 +80,9 @@ protected ProgressStep makeProgressStep(String input, Set<InputType> inputTypes)
       }
     }
 
-    throw new ValidationException(Collections.singletonList(new ValidationError("Could not convert input to any expected type.")));
+    throw new ValidationException(
+        Collections.singletonList(new ValidationError(String.format("'%s' does not match any of the expected types: (%s)",
```

Comment:
This is still not really meaningful. A "type" in this context is not a user term, and the input should match an object of one of the types - not a type itself.

Can add a method something like this to `LimsUtils` to help construct a more user-friendly message:

```
public String joinWithConjunction(Collection<String> words, String conjunction) {
    switch (words.size()) {
    case 0:
      throw new IllegalArgumentException("No words provided");
    case 1:
      return words.iterator().next();
    case 2:
      Iterator<String> iterator = words.iterator();
      return iterator.next() + " " + conjunction + " " + iterator.next();
    default:
      List<String> wordsCopy = new ArrayList<>(words);
      wordsCopy.set(wordsCopy.size() - 1, conjunction + " " + wordsCopy.get(wordsCopy.size() - 1));
      return String.join(", ", wordsCopy);
    }
  }
```

Reply:
Good point.  Thanks for the code, I will implement the suggestion to send errors similar to:
```
No Sample, Library, or Dilution found matching 'asdf'
```